runtime/runtime/src/prefetch.rs

```diff
@@ -346,8 +346,35 @@ mod tests {
             );
         }
 
-        // Request can still be pending. Stop threads and wait for them to finish.
-        std::thread::sleep(Duration::from_micros(1000));
+        // The queue is empty now but there can still be requests in progress.
+        // Looking at `Pending` slots in the prefetching area is not sufficient
+        // here because threads can be in `Trie::lookup` between nodes, at which
+        // point no slot is reserved for them but they are still doing work.
+        //
+        // Solution: drop `tries`, which also drops prefetchers. In particular,
+        // we want to drop the `PrefetchingThreadsHandle` stored in tries. This
+        // causes all background threads to stop after they finish the current
+        // work. It will even join them and wait until all threads are done.
+        //
+        // Because threads are joined, there is also a possibility this will
```

Comment:
nit: is it worth clarifying what `this` refers to here?  Maybe replace it with `drop(tries)`?